src/monitors/turbFlowStats.cpp

```diff
@@ -0,0 +1,268 @@
+                        turbPt[monitor->CatOffset.sum + p] += fieldPt[p];
+                        turbPt[monitor->CatOffset.sumSqr + p] += fieldPt[p] * fieldPt[p];
+                        turbPt[monitor->CatOffset.favreAvg + p] += turbPt[monitor->CatOffset.densityDtMult + p] / (turbPt[monitor->CatOffset.densityDtSum] + monitor->tiny);
+                        turbPt[monitor->CatOffset.rms + p] =
+                            PetscSqrtReal(turbPt[monitor->CatOffset.sumSqr + p] / (step + monitor->tiny) - PetscPowReal(turbPt[monitor->CatOffset.sum + p] / (step + monitor->tiny), 2));
+                        turbPt[monitor->CatOffset.mRms + p] = PetscSqrtReal(turbPt[monitor->CatOffset.densitySqr + p] / (turbPt[monitor->CatOffset.densitySum] + monitor->tiny) -
+                                                                            PetscPowReal(turbPt[monitor->CatOffset.densityMult + p] / (turbPt[monitor->CatOffset.densitySum] + monitor->tiny), 2));
+                    }
+                }
+            }
+        }
+        // Free the turbulent flow array
+        ierr = VecRestoreArray(monitor->turbVec, &turbDat);
+        CHKERRQ(ierr);
+    }
+    PetscFunctionReturn(0);
+}
+
+void ablate::monitors::TurbFlowStats::AddField(DM& dm, const char* nameField, PetscInt numComp) {
+    PetscFV fvm;
+    PetscFVCreate(PetscObjectComm(PetscObject(dm)), &fvm) >> checkError;
+    PetscObjectSetName((PetscObject)fvm, nameField) >> checkError;
+    PetscFVSetFromOptions(fvm) >> checkError;
+    PetscFVSetNumComponents(fvm, numComp) >> checkError;
+
+    DMAddField(dm, nullptr, (PetscObject)fvm) >> checkError;
+    PetscFVDestroy(&fvm);
+}
+
+void ablate::monitors::TurbFlowStats::Register(std::shared_ptr<ablate::solver::Solver> solverIn) {
+    // Copy over the solver
+    ablate::monitors::Monitor::Register(solverIn);
+
+    // Copy the master DM over, set fields, and initialize vector values
+    DMPlexFilter(this->GetSolver()->GetSubDomain().GetDM(), this->GetSolver()->GetSubDomain().GetLabel(), 1, &turbDM);
+    std::string DmName = "TurbDM";
+    PetscObjectSetName((PetscObject)turbDM, DmName.c_str());
+
+    // Add the densitySum and densityDtSum fields
+    std::string densitySum = "rhoSum";
+    AddField(turbDM, densitySum.c_str(), 1);
+
+    std::string densityDtSum = "rhoDtSum";
+    AddField(turbDM, densityDtSum.c_str(), 1);
+
+    // Initialize the numComps variable, which will store the sum of all the fields
+    PetscInt numComp = 0;
+
+    // Resize the fieldTrack and fieldComps vectors to be the proper size (fieldTrack one greater than num fields, fieldComps equal to num fields)
+    fieldComps.resize(fieldNames.size(), 0);
+    fieldTrack.resize(fieldNames.size() + 1, 0);
+    fieldTrack[0] = 0;
+
+    // Need to find the total number of field components and the indices marking the beginning of each field
+    for (int f = 0; f < (int)fieldNames.size(); f++) {
```

Comment:
Would you mind changing anything that uses .size to std::size_t instead of an int?